scripts/s3manage.py

```diff
@@ -0,0 +1,568 @@
+    finally:
+        f.close()
+
+    return data
+
+
+def save_to_s3(src, dest, cached=True, mimetype=None, break_on_error=True):
+
+    try:
+        with open(src, 'r') as f:
+            data = f.read()
+    except EnvironmentError:
+        print "Cannot upload {}".format(src)
+        if break_on_error:
+            sys.exit(2)
+        else:
+            return False
+    if mimetype is None:
+        mimetype, _ = mimetypes.guess_type(src)
+
+    _save_to_s3(data, dest, mimetype, cached=cached)
+
+
+def _save_to_s3(in_data, dest, mimetype, compress=True, cached=True):
+
+    data = in_data
+    compressed = False
+    content_encoding = None
+    cache_control = 'max-age=31536000, public'
+
+    extra_args = {}
+
+    if compress and mimetype not in NO_COMPRESS:
+        data = _gzip_data(in_data)
+        content_encoding = 'gzip'
+        compressed = True
+
+    print "Uploading to {} - {}, gzip: {}, cache headers: {}".format(dest, mimetype, compressed, cached)
+    if cached is False:
+        cache_control = 'no-cache, no-store, max-age=0, must-revalidate'
+
+    extra_args['ACL'] = 'public-read'
+    extra_args['ContentType'] = mimetype
+    extra_args['CacheControl'] = cache_control
+
+    try:
+        if compressed:
+            extra_args['ContentEncoding'] = content_encoding
+
+        if cached is False:
+            extra_args['Expires'] = datetime(1990, 1, 1)
+            extra_args['Metadata'] = {'Pragma': 'no-cache', 'Vary': '*'}
+
+        s3.Object(BUCKET_NAME, dest).put(Body=data, **extra_args)
+
+    except Exception as e:
+        print "Error while uploading {}: {}".format(dest, e)
+
+
+def get_index_version(c):
+    version = None
+    p = re.compile(ur'version: \'(\d+)\'')
+    match = re.findall(p, c)
+    if len(match) > 0:
+        version = int(match[0])
+
+    return version
+
+
+def usage():
+    print "\nManage map.geo.admin.ch versions in AWS S3 bucket\n"
+    print 'Usage: ' + os.path.basename(sys.argv[0]) + ' <command> [option]'
+    print
+    print "Commands:"
+    print
+    print "  list"
+    print "     list available <version> in bucket"
+    print
+    print "  upload [target] [dir]"
+    print "      upload content of /prd directory to bucket. You may specify"
+    print "      a directory (default to current)."
+    print "      Active project is NOT changed. Project has to be statified, "
+    print "      i.e. after PR https://github.com/geoadmin/mf-geoadmin3/pull/3078"
+    print
+    print "  activate <version>"
+    print "      activate the given 'version', i.e."
+    print "      copy from <version>/index.<version>.html to index.html"
+    print
+    print "  info <version>"
+    print "      print build info on 'version' (branch, build date, git hash...)"
+    print
+    print "  delete <version>"
+    print "      delete the given 'version' (both directory and indexes files"
+    print
+    print "  help"
+    print "      print this message"
+
+
+def upload(version, base_dir):
```

Comment:
The whole function is a bit too sensitive I think and we will need to remember to adapt this script if we change the project structure and so on.
Can't we:

1. The mapping between the extension and the mimetype
2. What we want to always exclude
3. What should not be cached.

And then simply `os.walk` through the `prd` directory.

According to the staging `dev` and `int` for instance always set UPLOAD_SRC_DIR to True and if staging is `prod` set it to False.